lib/middleware.js

```diff
@@ -347,6 +348,42 @@ class Kevin {
                 return;
             }
 
+            if (req.method === "POST" && reqPath === `${this.kevinApiPrefix}/restart-compiler`) {
+                // This endpoint restarts compilers. It accepts two query params:
+                // `compiler` (required), which is the name of the compiler, and
+                // `hard` (optional) which, if true, fully shuts down and restarts the compiler,
+                //          rather than just invalidating it and forcing a partial recompilation.
+                if (!req.query.compiler || !manager.isCompilerActive(req.query.compiler)) {
+                    return res.status(400).send("Kevin couldn't find a compiler with that name");
+                }
+                // If hard is set to anything that looks like "true", let's force restart the compiler.
+                if (req.query.hard && req.query.hard.toLowerCase() === "true") {
+                    manager.closeCompiler(req.query.compiler)
+                        .then((name) => {
+                            if (!name) {
+                                res.status(400).send("Kevin couldn't find a compiler with that name");
```

Comment:
It seems like this case would cause you to do `res.status(400).send("...")` but then still doing `return this.buildConfig` and then the `res.sendStatus(200)` call.

Reply:
Nope, that's a missing return statement. It also looks like `buildCompiler` doesn't return when its given a false name, even though it detects when a name is invalid. Good catch!